core/pkg/service/ofrep/models.go

```diff
@@ -0,0 +1,106 @@
+package ofrep
+
+import (
+	"fmt"
+
+	"github.com/open-feature/flagd/core/pkg/evaluator"
+	"github.com/open-feature/flagd/core/pkg/model"
+)
+
+type Request struct {
+	Context interface{} `json:"context"`
+}
+
+type EvaluationSuccess struct {
+	Value    interface{} `json:"value"`
+	Key      string      `json:"key"`
+	Reason   string      `json:"reason"`
+	Variant  string      `json:"variant"`
+	Metadata interface{} `json:"metadata"`
+}
+
+type BulkEvaluationResponse struct {
+	Flags []interface{} `json:"flags,omitempty"`
+}
+
+type EvaluationError struct {
+	Key          string `json:"key"`
+	ErrorCode    string `json:"errorCode"`
+	ErrorDetails string `json:"errorDetails"`
+}
+
+type BulkEvaluationError struct {
+	ErrorCode    string `json:"errorCode"`
+	ErrorDetails string `json:"errorDetails"`
+}
+
+type InternalError struct {
+	ErrorDetails string `json:"errorDetails"`
+}
+
+func BulkEvaluationResponseFrom(values []evaluator.AnyValue) BulkEvaluationResponse {
+	evaluations := make([]interface{}, 0)
+
+	for _, value := range values {
+		if value.Error != nil {
+			_, evaluationError := ValueToStatusAndError(value)
+			evaluations = append(evaluations, evaluationError)
+		} else {
+			evaluations = append(evaluations, SuccessResponseFrom(value))
+		}
+	}
+
+	return BulkEvaluationResponse{
+		evaluations,
+	}
+}
+
+func SuccessResponseFrom(result evaluator.AnyValue) EvaluationSuccess {
+	return EvaluationSuccess{
+		Value:    result.Value,
+		Key:      result.FlagKey,
+		Reason:   result.Reason,
+		Variant:  result.Variant,
+		Metadata: result.Metadata,
+	}
+}
+
+func ContextErrorResponseFrom(key string) EvaluationError {
+	return EvaluationError{
+		Key:          key,
+		ErrorCode:    model.InvalidContextCode,
+		ErrorDetails: "provided context is invalid",
+	}
+}
+
+func ContextErrorResponseForBulkAndStatus() BulkEvaluationError {
+	return BulkEvaluationError{
+		ErrorCode:    model.InvalidContextCode,
+		ErrorDetails: "provided context is invalid",
+	}
+}
+
+func ValueToStatusAndError(result evaluator.AnyValue) (int, EvaluationError) {
```

Comment:
I would maybe change the name to `ErrorResponseFrom` (similar to the prev function) or something more descriptive that we should call this function only when an error occurs

Reply:
Done, renamed to `EvaluationErrorResponseFrom`